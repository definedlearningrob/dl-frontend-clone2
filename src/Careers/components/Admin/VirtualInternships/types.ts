import { TOpportunity as TBaseOpportunity } from '@dc/resources/types';

export const enum VirtualInternshipStatuses {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export type VirtualInternshipLessonAttributes = {
  lessonId: string;
  step: number;
};

export type FormValues = {
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  description: string;
  name: string;
  pathways: {
    label: string;
    value: string;
  }[];
  availableSpots: number;
  creditsOutcomes: string;
  image: Record<string, unknown>;
  tags: {
    label: string;
    value: string;
  }[];
  requiredExperiences: number;
  status: { value: VirtualInternshipStatuses; label: string };
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};

export type TPathway = Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>;

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'opportunityType'
  | 'tags'
  | 'salaryInformation'
> & {
  pathways: TPathway[];
};

export type Lesson = {
  id: string;
  step: number;
  name: string;
  imageUrl: string;
  thumbnailUrl: string;
  type: string;
};

export type TVirtualInternship = {
  archivedAt: string | null;
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  id: string;
  opportunity: TOpportunity;
  requiredExperiences: number;
  status: VirtualInternshipStatuses;
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};
