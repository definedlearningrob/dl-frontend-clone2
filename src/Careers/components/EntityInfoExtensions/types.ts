export type TAdminEntityInfoExtensionField = {
  id: string;
  name: string;
};
