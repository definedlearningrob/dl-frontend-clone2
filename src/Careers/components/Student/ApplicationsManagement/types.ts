export type Counselor = {
  email: string;
  firstName: string;
  fullName: string;
  lastName: string;
  name: string;
  username: string;
  uuid: string;
};

export type AvailableTeacher = {
  email: string;
  fullName: string;
  name: string;
  username: string;
  uuid: string;
};

export type CounselorSelectOption = { label: string; value: string };
