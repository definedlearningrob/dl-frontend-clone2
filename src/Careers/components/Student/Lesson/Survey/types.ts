export type SURVEY_QUESTION_TYPE = 'single_choice' | 'multiple_choice';

type CareerReviewSurveyQuestionOption = {
  option: string;
  step: number;
};

type CareerReviewSurveyQuestion = {
  answer: string[];
  id: string;
  options: CareerReviewSurveyQuestionOption[];
  question: string;
  type: SURVEY_QUESTION_TYPE;
};

export type CareerReviewSurvey = {
  __typename: string;
  questions: CareerReviewSurveyQuestion[];
  performed: boolean;
  version: number;
};
