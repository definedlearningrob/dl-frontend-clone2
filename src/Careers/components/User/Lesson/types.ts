import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

import {
  TCheckInGroup as TBaseCheckInGroup,
  TCheckInQuestion as TBaseCheckInQuestion,
} from '@shared/components/CheckIns/types';

type TRubricHeading = {
  id: string;
  multiplier: number;
  name: string;
  uuid: string;
};

type TRubricCriteria = {
  id: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
  uuid: string;
};

type TRubricCriteriaLabel = {
  displayName: string;
  id: string;
  score: number;
  uuid: string;
};

type TRubric = {
  id: string;
  uuid: string;
  name: string;
  description: string;
  headings: TRubricHeading[];
  criterias: TRubricCriteria[];
  criteriaLabels: TRubricCriteriaLabel[];
};

type TAssignmentSubmissionFile = {
  filename: string;
  googleWeblink?: string;
  id: string;
  source: string;
  url: string;
};

type TSubmissionGrade = {
  id: string;
  lastGradedBy: {
    uuid: string;
    firstName: string;
    lastName: string;
    fullName: string;
  };
  status: 'ACCEPTED' | 'NOT_ACCEPTED';
  createdAt: string;
  updatedAt: string;
};

export type TAssignmentSubmission = {
  files: TAssignmentSubmissionFile[];
  grade: TSubmissionGrade | null;
  id: string;
  status: 'DRAFT' | 'SUBMITTED';
};

type TAttachmentFile = {
  filename: string;
  id: string;
  url: string;
};

type TExtensionField = {
  archivedAt: string | null;
  author: {
    email: string;
    firstName: string;
    lastName: string;
    username: string;
    uuid: string;
  };
  description: string;
  files: {
    filename: string;
    id: string;
    url: string;
  }[];
  id: string;
  imageUrl: string;
  links: {
    name: string;
    url: string;
  }[];
  name: string;
};

type TBadge = {
  archivedAt: string | null;
  id: string;
  imageUrl: string;
  name: string;
};

export type TAssignment = {
  archivedAt: string | null;
  assetName: string;
  badges: TBadge[];
  createdAt: string;
  id: string;
  name: string;
  description: string;
  displayName: string;
  step: number;
  rubrics: TRubric[];
  submission: TAssignmentSubmission | null;
  lessons: TLesson[];
  __typename: 'Assignment';
};

export type TAttachment = {
  archivedAt: string | null;
  createdAt: string;
  description: string;
  displayName: string;
  files: TAttachmentFile[];
  id: string;
  step: number;
  lessons: TLesson[];
  name: string;
  __typename: 'Attachment';
};

export type TCareerReviewSurvey = {
  questions: {
    answer: string[];
    id: string;
    options: {
      option: string;
      step: number;
    }[];
    question: string;
    type: SURVEY_QUESTION_TYPE;
  }[];
};

export type TExternalPresentation = {
  archivedAt: string | null;
  displayName: string;
  id: string;
  source: string;
  step: number;
  isExpandable: boolean;
  __typename: 'ExternalPresentation';
};

export type TResearchLink = {
  archivedAt: string | null;
  author: string;
  createdAt: string;
  displayName: string;
  id: string;
  resourceLink: string;
  sourceName: string;
  step: number;
  __typename: 'ResearchLink';
};

export type TText = {
  archivedAt: string | null;
  content: string;
  createdAt: string;
  displayName: string;
  id: string;
  step: number;
  __typename: 'Text';
};

export type TVideo = {
  archivedAt: string | null;
  createdAt: string;
  description: string;
  displayName: string;
  filename: string;
  id: string;
  step: number;
  url: string;
  __typename: 'Video';
};

export type TVocabulary = {
  archivedAt: string | null;
  createdAt: string;
  definition: string;
  id: string;
  step: number;
  term: string;
  __typename: 'Vocabulary';
};

export type TCheckInQuestion = Pick<
  TBaseCheckInQuestion,
  'id' | 'question' | 'step' | 'answer' | '__typename'
>;

export type TCheckInGroup = Omit<TBaseCheckInGroup, 'questions'> & {
  questions: TCheckInQuestion[];
};

export type TLesson = {
  archivedAt: string | null;
  id: string;
  imageUrl: string;
  name: string;
  assignments: TAssignment[];
  attachments: TAttachment[];
  badges: TBadge[];
  careerReviewSurvey: TCareerReviewSurvey | null;
  checkInGroups: TCheckInGroup[];
  checkInQuestions: TCheckInQuestion[];
  description: {
    audience: string;
    goal: string;
    introduction: string;
    role: string;
    situation: string;
  };
  extensionFields: TExtensionField[];
  externalPresentations: TExternalPresentation[];
  hasPresentation: boolean;
  researchLinks: TResearchLink[];
  step: number;
  texts: TText[];
  thumbnailUrl: string;
  type: string;
  videos: TVideo[];
  vocabularies: TVocabulary[];
};
