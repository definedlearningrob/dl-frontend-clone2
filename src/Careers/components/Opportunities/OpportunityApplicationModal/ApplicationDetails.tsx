import { useTranslation } from 'react-i18next';
import cx from 'classnames';

import { TOpportunity } from '@dc/resources/types';

import DefaultImage from '@shared/assets/images/default-thumbnail.svg';
import { ReactComponent as EducatorIcon } from '@shared/svg/educator.svg';
import { ReactComponent as CalendarIcon } from '@shared/svg/calendar.svg';
import { ReactComponent as DeadlineIcon } from '@shared/svg/deadline.svg';
import { ReactComponent as LocationIcon } from '@shared/svg/location-marker.svg';
import SharedImage from '@shared/components/Image/Image';
import TruncatedText from '@shared/components/TruncatedText';
import SharedIcon from '@shared/components/Icon/Icon';
import { formatDateTime, getPeriod } from '@shared/utils/date/date';
import { cleanInjection } from '@shared/utils/cleanInjection';

import styles from './ApplicationDetails.module.sass';

type Props = {
  opportunity: Pick<
    TOpportunity,
    | 'imageUrl'
    | 'name'
    | 'description'
    | 'opportunityType'
    | 'periodEnd'
    | 'periodStart'
    | 'deadline'
    | 'location'
  >;
};

export const ApplicationDetails = ({ opportunity }: Props) => {
  const { t } = useTranslation();
  const imageUrl = opportunity.imageUrl || '';
  const periodValue =
    opportunity.periodStart && opportunity.periodEnd
      ? getPeriod(opportunity.periodStart, opportunity.periodEnd)
      : 'N/A';
  const deadlineValue = opportunity.deadline ? formatDateTime(opportunity.deadline) : 'N/A';

  const detailsClasses = cx(
    styles.opportunityDetails,
    'scrollbar overflow-y-auto overflow-x-hidden'
  );

  return (
    <div className={detailsClasses}>
      <section className={styles.opportunityDescription}>
        <SharedImage
          className={styles.descriptionImage}
          fallbackSrc={DefaultImage}
          src={imageUrl}
        />
        <h5 className={styles.descriptionTitle}>{opportunity.name}</h5>
        <TruncatedText text={opportunity.description} />
      </section>
      <section className={styles.opportunityInfo}>
        <span className={styles.opportunityInfoWrapper}>
          <SharedIcon icon={<EducatorIcon />} size='xs' />
          <span className={styles.infoLabel}>{t('opportunities.type')}</span>
          {t(`opportunities.types.${opportunity.opportunityType}`)}
        </span>
        <span className={styles.opportunityInfoWrapper}>
          <SharedIcon icon={<CalendarIcon />} size='xs' />
          <span className={styles.infoLabel}>{t('opportunities.period')}</span>
          {periodValue}
        </span>
        <span className={styles.opportunityInfoWrapper}>
          <SharedIcon icon={<DeadlineIcon />} size='xs' />
          <span className={styles.infoLabel}>{t('opportunities.deadline')}</span>
          {deadlineValue}
        </span>
        {opportunity.location && (
          <span className={styles.opportunityInfoWrapper}>
            <SharedIcon icon={<LocationIcon />} size='xs' />
            <span className={styles.infoLabel}>{t('opportunities.location')}</span>
            {/* eslint-disable-next-line react/no-danger */}
            <div dangerouslySetInnerHTML={cleanInjection(opportunity.location)} />
          </span>
        )}
      </section>
    </div>
  );
};
