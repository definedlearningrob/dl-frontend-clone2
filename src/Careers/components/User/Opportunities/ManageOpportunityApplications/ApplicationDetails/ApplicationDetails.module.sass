@import '@shared/stylesheets/config.sass'

$action-button-width: 160px

.details
  +scrollbar
  height: 100%
  padding-right: $m-base
  margin-bottom: $m-base
  flex: 1 1 auto

  +media-breakpoint-up(xxxl)
    padding-right: $m-md

.actionsWrapper
  flex: 0 0
  height: 100%

.content
  font-size: $font-size-xs
  line-height: $line-height-lg
  margin-bottom: $m-base

  +media-breakpoint-up(xxxl)
    font-size: $font-size-sm

.actionButton
  width: $action-button-width !important

.rateInfoWrapper
  +flex-container(row, false, flex-start)
  column-gap: $m-xs
  border: 1px solid $color-neutral-300
  border-radius: $m-xs
  padding: $m-xs

  +media-breakpoint-up(xxxl)
    column-gap: $m-sm
    padding: $m-base

.statusIconWrapper
  padding: $m-xs
  border-radius: 100%

.accepted
  color: $color-success-500
  background-color: $color-success-100

.rejected
  color: $color-danger-500
  background-color: $color-danger-100

.rateInfoDescription
    color: $color-neutral-700
    font-size: $font-size-xs
    line-height: $line-height-lg
    margin-bottom: 0

    +media-breakpoint-up(xxxl)
        font-size: $font-size-sm
