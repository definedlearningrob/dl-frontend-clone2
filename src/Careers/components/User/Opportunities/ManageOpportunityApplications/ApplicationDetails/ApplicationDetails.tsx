import { useTranslation } from 'react-i18next';
import { useCallback } from 'react';

import { ReactComponent as ThumbsUpIcon } from '@dc/assets/icons/thumbs-up.svg';
import { ReactComponent as ThumbsDownIcon } from '@dc/assets/icons/thumbs-down.svg';
import {
  AcceptedStatuses,
  useUpdateOpportunityApplication,
} from '@dc/graphql/user/hooks/useUpdateOpportunityApplication';
import { OPPORTUNITY_APPLICATION_STATUS } from '@dc/resources/enums';
import { TOpportunityApplications } from '@dc/graphql/user/queries/opportunityApplications';
import { ApplicationRatingInfo } from '@dc/components/User/Opportunities/ManageOpportunityApplications/ApplicationRatingInfo/ApplicationRatingInfo';
import { TOpportunityApplication } from '@dc/resources/types';

import useQueryParams from '@shared/hooks/useQueryParams';
import SharedButton from '@shared/components/Button/Button';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

import styles from './ApplicationDetails.module.sass';

type Props = {
  opportunity: TOpportunityApplications;
  application: TOpportunityApplication;
};

export const ApplicationDetails = ({ opportunity, application }: Props) => {
  const { params } = useQueryParams<{ applicationId: string }>();
  const { updateApplicationStatus } = useUpdateOpportunityApplication();
  const isDesktop = useBreakpointUp({ breakpoint: 'xxxl' });
  const { t } = useTranslation();

  const questions = opportunity.questions;

  if (!application) return null;

  const applicationDetails = questions.map(({ question, id }) => {
    const answer = application.answers.find(
      ({ opportunityQuestionId }) => opportunityQuestionId === id
    );

    return { question, answer: answer?.answer || '' };
  });

  const handleStatusChange = (status: AcceptedStatuses) => {
    updateApplicationStatus(params.applicationId, status);
  };

  const handleResetApplication = useCallback(() => {
    handleStatusChange(OPPORTUNITY_APPLICATION_STATUS.PENDING);
  }, [handleStatusChange]);

  const buttonSize = isDesktop ? 'lg' : 'md';

  const hasBeenRated = [
    OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
    OPPORTUNITY_APPLICATION_STATUS.REJECTED,
  ].includes(application.status);

  return (
    <div className='flex flex-col h-full'>
      <div className={styles.details}>
        {applicationDetails.map(({ question, answer }) => (
          <div key={`question-${answer}`}>
            <h5 className='text-neutral-800 text-xs font-bold mb-xs leading-base xxxl:text-sm'>
              {question}
            </h5>
            <p className='text-neutral-700 text-xxs leading-lg mb-base xxxl:text-sm'>{answer}</p>
          </div>
        ))}
      </div>
      <div className='basis-0 pe-base xxxl:pe-md'>
        {!hasBeenRated && (
          <div className='flex gap-xs xxxl:gap-sm'>
            <SharedButton
              Icon={ThumbsUpIcon}
              className={styles.actionButton}
              size={buttonSize}
              variant='primary'
              onClick={() => handleStatusChange(OPPORTUNITY_APPLICATION_STATUS.ACCEPTED)}>
              {t('opportunityManageApplications.details.accept')}
            </SharedButton>
            <SharedButton
              Icon={ThumbsDownIcon}
              className={styles.actionButton}
              size={buttonSize}
              variant='danger'
              onClick={() => handleStatusChange(OPPORTUNITY_APPLICATION_STATUS.REJECTED)}>
              {t('opportunityManageApplications.details.reject')}
            </SharedButton>
          </div>
        )}
        {hasBeenRated && (
          <div className={styles.rateInfoWrapper}>
            <ApplicationRatingInfo
              application={application}
              onResetApplication={handleResetApplication}
              onStatusChange={handleStatusChange}
            />
          </div>
        )}
      </div>
    </div>
  );
};
