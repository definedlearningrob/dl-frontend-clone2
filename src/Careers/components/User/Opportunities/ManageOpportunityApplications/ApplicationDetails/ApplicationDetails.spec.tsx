import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Route } from 'react-router-dom';
import { OpportunityTypes } from '@graphql/dc/shared/types';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { ApplicationDetails } from '@dc/components/User/Opportunities/ManageOpportunityApplications/ApplicationDetails/ApplicationDetails';
import { OPPORTUNITY_APPLICATION_STATUS } from '@dc/resources/enums';
import {
  TOpportunityApplicationData,
  TOpportunityApplicationVariables,
  UPDATE_OPPORTUNITY_APPLICATION,
} from '@dc/graphql/user/mutations/updateOpportunityApplication';
import cacheConfig from '@dc/graphql/cacheConfig';

const application = {
  id: '1',
  appliedAt: '2023-05-01',
  updatedAt: '2023-05-02',
  status: OPPORTUNITY_APPLICATION_STATUS.PENDING,
  answers: [{ id: '1', opportunityQuestionId: '1', answer: 'To sleep better!' }],
  student: {
    uuid: '123',
    fullName: 'Applicant Student',
    schoolClasses: [{ uuid: 'class-uuid', name: 'Class name' }],
  },
};

const opportunity = {
  hasPendingApplications: true,
  questions: [{ id: '1', question: 'Why you want to test this?', answer: 'To sleep better!' }],
  opportunityType: OpportunityTypes.APPRENTICESHIP,
  automaticAcceptance: false,
  id: '1',
  applications: { nodes: [application], pagesCount: 1, nodesCount: 1 },
  filteredApplications: { nodes: [], pagesCount: 1, nodesCount: 0 },
  name: 'Opportunity name',
};

const acceptSpy = jest.fn();
const rejectSpy = jest.fn();

const acceptApplicationStatusMock: MockedResponse<
  TOpportunityApplicationData,
  TOpportunityApplicationVariables
> = {
  request: {
    query: UPDATE_OPPORTUNITY_APPLICATION,
    variables: { input: { id: '1', status: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED } },
  },
  result() {
    acceptSpy();

    return {
      data: {
        updateOpportunityApplication: {
          application: {
            status: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
            lastChangedBy: { name: 'Cindy Amino' },
          },
        },
      },
    };
  },
};

const rejectApplicationStatusMock: MockedResponse<
  TOpportunityApplicationData,
  TOpportunityApplicationVariables
> = {
  request: {
    query: UPDATE_OPPORTUNITY_APPLICATION,
    variables: { input: { id: '1', status: OPPORTUNITY_APPLICATION_STATUS.REJECTED } },
  },
  result() {
    rejectSpy();

    return {
      data: {
        updateOpportunityApplication: {
          application: {
            status: OPPORTUNITY_APPLICATION_STATUS.REJECTED,
            lastChangedBy: { name: 'Cindy Amino' },
          },
        },
      },
    };
  },
};

describe('ApplicationDetails', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <MockedProvider>
        <ApplicationDetails application={application} opportunity={opportunity} />
      </MockedProvider>
    );

    expect(screen.getByText('Why you want to test this?')).toBeInTheDocument();
    expect(screen.getByText('To sleep better!')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Accept' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Reject' })).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should allow accepting an application', async () => {
    renderWithRouterAndReduxProvider(
      <Route path='/manage-opportunities'>
        <MockedProvider cache={cacheConfig} mocks={[acceptApplicationStatusMock]}>
          <ApplicationDetails application={application} opportunity={opportunity} />
        </MockedProvider>
      </Route>,
      { route: '/manage-opportunities?applicationId=1' }
    );

    userEvent.click(screen.getByRole('button', { name: 'Accept' }));
    await waitFor(() => {
      expect(acceptSpy).toHaveBeenCalledTimes(1);
    });
  });

  it('should allow rejecting an application', async () => {
    renderWithRouterAndReduxProvider(
      <Route path='/manage-opportunities'>
        <MockedProvider cache={cacheConfig} mocks={[rejectApplicationStatusMock]}>
          <ApplicationDetails application={application} opportunity={opportunity} />
        </MockedProvider>
      </Route>,
      { route: '/manage-opportunities?applicationId=1' }
    );

    userEvent.click(screen.getByRole('button', { name: 'Reject' }));

    await waitFor(() => {
      expect(rejectSpy).toHaveBeenCalledTimes(1);
    });
  });
});
