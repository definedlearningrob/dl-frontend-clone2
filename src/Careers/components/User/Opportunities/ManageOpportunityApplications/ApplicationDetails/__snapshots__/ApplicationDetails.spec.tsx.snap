// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDetails should render correctly 1`] = `
<div>
  <div
    class="flex flex-col h-full"
  >
    <div
      class="details"
    >
      <div>
        <h5
          class="text-neutral-800 text-xs font-bold mb-xs leading-base xxxl:text-sm"
        >
          Why you want to test this?
        </h5>
        <p
          class="text-neutral-700 text-xxs leading-lg mb-base xxxl:text-sm"
        >
          To sleep better!
        </p>
      </div>
    </div>
    <div
      class="basis-0 pe-base xxxl:pe-md"
    >
      <div
        class="flex gap-xs xxxl:gap-sm"
      >
        <button
          class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] actionButton -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
          data-testid="button"
          type="button"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          <span
            class="items-center flex justify-center gap-xs"
          >
            Accept
          </span>
        </button>
        <button
          class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] actionButton -danger bg-danger-500 hover:bg-danger-600 outline-danger-500 border-danger-500 py-xs px-base leading-[22px] border text-white"
          data-testid="button"
          type="button"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          <span
            class="items-center flex justify-center gap-xs"
          >
            Reject
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
